src/app/store/effects/VehicleTestResultModel.effects.spec.ts

```diff
@@ -1,62 +1,154 @@
-import { Observable, ReplaySubject } from 'rxjs';
-import { TechnicalRecordServiceMock } from '../../../../testconfig/services-mocks/technical-record-service.mock';
+import { BehaviorSubject, Observable, of } from 'rxjs';
 import { TestBed } from '@angular/core/testing';
 import { RouterTestingModule } from '@angular/router/testing';
-import { INITIAL_STATE, Store } from '@ngrx/store';
-import { hot } from 'jasmine-marbles';
-import { provideMockActions } from '@ngrx/effects/testing';
+import { Action, Store } from '@ngrx/store';
+import { cold, hot } from 'jasmine-marbles';
 import {
   GetVehicleTestResultModel,
-  GetVehicleTestResultModelSuccess
+  GetVehicleTestResultModelSuccess,
+  SetCurrentState,
+  UpdateTestResult,
+  UpdateTestResultSuccess
 } from '@app/store/actions/VehicleTestResultModel.actions';
 import { IAppState } from '@app/store/state/app.state';
 import { VehicleTestResultModelEffects } from '@app/store/effects/VehicleTestResultModel.effects';
 import { TestResultService } from '@app/technical-record-search/test-result.service';
-import { TestResultModel } from '@app/models/test-result.model';
+import { provideMockActions } from '@ngrx/effects/testing';
+import { TestResultTestTypeNumber } from '@app/models/test-result-test-type-number';
+import { TESTING_TEST_MODELS_UTILS } from '@app/utils/testing-test-models.utils';
+import { VIEW_STATE } from '@app/app.enums';
+import { ClearErrorMessage } from '@app/store/actions/Error.actions';
+import { getVehicleTestResultModel } from '@app/store/selectors/VehicleTestResultModel.selectors';
+import { map } from 'rxjs/operators';
+import { UserService } from '@app/app-user.service';
+import { UserDetails } from '@app/models/user-details';
+
+const testResult = [TESTING_TEST_MODELS_UTILS.mockTestRecord()];
+const mockSelector = new BehaviorSubject<any>(undefined);
 
-const testResult = {} as TestResultModel;
+class MockStore {
```

Comment:
Can we move this into the utils/ folder? I think we might even one in there already in fact?
We can import in this file and use the function if it is already in there. This is likely to be reused and extended as the features come so that would make more sense to have it in an utils than this file I think.